### PR TITLE
Add reason field #21, and fix bug in disapproval method

### PR DESCRIPTION
**Pull request by nagi1**
I found a bug in disapproval method, when disapprove a creating action, modifer_id will be null so `applyModificationChanges()` will be called on null.

I copied the same implementation from `approve()` method, and added to tests.

### CODE REVIEW

**Comment by nagi1** [comment]
Please merge this as soon as you get this. Thanks.

**Comment by stephenlake** [comment]
Thanks for the contribution, but I have to decline this PR:
- The lock should never be ignored.
- Style guides have been tainted (Files that do not need to be changed have been & defaulting string values are a nogo).

Please see https://www.php-fig.org/

File: docs/README.md [hunk 1]
```diff
@@ -276,6 +287,15 @@ $approval     = $modification->disapprovals()->get();
 $author       = $approval->disapprover();
 ```
 
+## Retrieving disapproval Reason
+
+```php
+$post         = Post::find(1);
+$modification = $post->modifications()->first();
+$approval     = $modification->disapprovals()->get();
+$reason       = $approval->reason;
```
Comment (stephenlake) [inline docs/README.md:296]: Right off the bat without testing, how would this work? `->get()` returns a collection, calling `->reason` on a collection would most likely always return null.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /vendor
+composer.lock
```
Comment (stephenlake) [inline .gitignore]: Please remove, without the lock present the package manager may install incompatible dependency versions.

File: src/Traits/ApprovesChanges.php [hunk 3]
```diff
@@ -78,11 +80,12 @@ public function approve(\Approval\Models\Modification $modification) : bool
     /**
      * Disapprove a modification.
      *
+     * @param string|null $reason
```
Comment (stephenlake) [inline src/Traits/ApprovesChanges.php]: Bad parameter ordering.

**Comment by stephenlake** [comment]
Thanks for the contribution:
- The lock should never be ignored.
- CS standards have been tainted.

Please see https://www.php-fig.org/ - If you can fix these minor issues I'll gladly patch it with your addition, thank you for updating the readme and adding tests, you're a gem :gem:

File: docs/README.md (hunk 1 above)
Comment (nagi1) [reply docs/README.md:296]: I would use `->first()`

File: .gitignore (hunk 2 above)
Comment (nagi1) [reply .gitignore]: No problem

**Comment by nagi1** [comment]
Thanks for checking out the commit, I fix it.
please merge because I'm using my forked branch.

**Pull request merged**